Hazel/premake5.lua

```diff
@@ -43,12 +43,32 @@ project "Hazel"
 		"GLFW",
 		"Glad",
 		"ImGui",
-		"opengl32.lib"
 	}
 
+	filter "system:linux"
+		-- pic "on"
+
+		links 
+		{ 
+			"Xrandr",
+			"Xi",
+			"GLU",
+			"GL",
+			"X11"
+		}
+
+		defines
+		{
+		}
+
 	filter "system:windows"
 		systemversion "latest"
 
+		links 
+		{ 
+			"opengl32.lib"
+		}
+		
 		defines
```

Comment:
Premake should handle the platform specific defines. `HZ_PLATFORM_WINDOWS` should be here otherwise there isn't much point in having Premake.

Reply:
Come to think of it. Premake handles every project submodule and it specifies the platform type for these libraries. A top down approach would make the entire project much neater.

Reply:
Cherno did say that the current way of finding the platform (using predefined macros) in [`Hazel/Core/PlatformDetection.h`](https://github.com/TheCherno/Hazel/blob/857d8d60e63b23c67edf1bb10e2715fb57089597/Hazel/src/Hazel/Core/PlatformDetection.h) was indeed somewhat an issue. This also feels out of the scope of this PR.. In #320 this was changed to a seperate file and added to the pch...